src/unionalgorithms.jl

```diff
@@ -0,0 +1,67 @@
+import TupleTools
+import StaticArrays: SVector
+
+"""
+    leftof(I1::TypedEndpointsInterval, I2::TypedEndpointsInterval)
+
+Returns if `I1` has a part to the left of `I2`.
+"""
+function leftof(I1::TypedEndpointsInterval{L1,R1}, I2::TypedEndpointsInterval{L2,R2}) where {L1,R1,L2,R2}
+    if leftendpoint(I1) < leftendpoint(I2)
+        true
+    elseif leftendpoint(I1) > leftendpoint(I2)
+        false
+    elseif L1 == :closed && L2 == :open
+        true
+    else
+        false
+    end
+end
+
+"""
+    canunion(d1, d2)
+
+Returns if `d1 ∪ d2` is a single interval.
+"""
+@inline canunion(d1, d2) = any(∈(d1), endpoints(d2)) || any(∈(d2), endpoints(d1))
```

Comment:
Do we need this function?
It's just `canunion(a, b) = !isdisjoint(a, b)`.

Reply:
i copied it from the old union function. I think they are different because an open and a closed interval can be disjoint while have a connected union.